File: dfs_bfs/programmers/practice_kit_1.py
def dfs(i, sum, numbers, target):
    answer = 0
    
    if i == len(numbers):
        # 끝 부분에 도달하였을 때 합계가 target이면
        if sum == target:
            return 1
        return 0

    answer += dfs(i + 1, sum + numbers[i], numbers, target)
    answer += dfs(i + 1, sum - numbers[i], numbers, target)
    
    return answer

def solution(numbers, target):
    answer = 0
    answer += dfs(0, 0, numbers, target)
    return answer

File: dfs_bfs/programmers/practice_kit_1.java
package dfs_bfs.programmers;


public class practice_kit_1 {
    static int[] numbers = {1, 1, 1, 1, 1};
    static int target = 3;

    public static void main(String[] args) {
        Solution solu = new Solution();
        System.out.println(solu.solution(numbers, target));
    }

    static class Solution {
        public int dfs(int i, int sum, int[] numbers, int target) {
            int answer = 0;

            if (i == numbers.length) {
                if (sum == target) return 1;
                return 0;
            }

            answer += dfs(i + 1, sum + numbers[i], numbers, target);
            answer += dfs(i + 1, sum - numbers[i], numbers, target);

            return answer;
        }

        public int solution(int[] numbers, int target) {
            int answer = 0;
            answer += dfs(0, 0, numbers, target);
            return answer;
        }
        
    }
    
}
